src/screens/LocalCommunity/index.js

```diff
@@ -163,10 +192,28 @@ const LocalCommunity = () => {
   const onPressFirstLocation = (status) => {
     setIsVisibleFirstLocation(status);
     setSearch('');
+    if (status) {
+      AnalyticsEventTracking.eventTrack(
+        BetterSocialEventTracking.ONBOARDING_LOCATION_FIRST_OPEN_SEARCH
+      );
+    } else if (!status && !location?.at(0)) {
+      AnalyticsEventTracking.eventTrack(
+        BetterSocialEventTracking.ONBOARDING_LOCATION_DRAWER_CLOSED
+      );
+    }
   };
 
   const onPressSecondLocation = (status) => {
     setIsVisibleSecondLocation(status);
+    if (status) {
+      AnalyticsEventTracking.eventTrack(
+        BetterSocialEventTracking.ONBOARDING_LOCATION_FIRST_OPEN_SEARCH
+      );
+    } else if (!status && !location?.at(1)) {
+      AnalyticsEventTracking.eventTrack(
+        BetterSocialEventTracking.ONBOARDING_LOCATION_DRAWER_CLOSED
+      );
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This code tracks when the search for the first and second locations is opened and when the drawer is closed without a location being selected. If it's necessary to track these events for all locations, consider generalizing this logic.
```diff
+    const onPressNthLocation = (status, index) => {
+      setIsVisibleNthLocation(status);
+      if (status) {
+        AnalyticsEventTracking.eventTrack(
+          BetterSocialEventTracking.ONBOARDING_LOCATION_NTH_OPEN_SEARCH
+        );
+      } else if (!status && !location?.at(index)) {
+        AnalyticsEventTracking.eventTrack(
+          BetterSocialEventTracking.ONBOARDING_LOCATION_DRAWER_CLOSED
+        );
+      }
+      setSearch('');
+    };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->